tests/requests.py

```diff
@@ -0,0 +1,106 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Callable, Optional, Union
+
+import pytest
+
+
+class Url(Enum):
+    NORMAL = ("/v1/wfs/",)
+    FLAT = ("/v1/wfs-flattened/",)
+    COMPLEX = ("/v1/wfs-complextypes/",)
+    GENERATED = ("/v1/wfs-gen-field/",)
+
+    def __init__(self, url):
+        self.url = url
+
+
+@dataclass
+class Request:
+    method: str
+    _url: Url
+    id: Optional[str]
+    expect: Optional[Any]
+
+    def __init__(self, *, id=None, expect=None, url_type: str = "NORMAL"):
+        self.id = id
+        self.expect = expect
+        self._url = Url[url_type]
+
+    def test_id(self):
+        if self.id:
+            return f"{self.method}-{self.id} ({self._url.name})"
+        return f"{self.method} ({self._url.name})"
+
+    @property
+    def url(self):
+        return self._url.url
+
+    @url.setter
+    def url(self, url_type):
+        self._url = Url[url_type]
+
+    @property
+    def url_type(self):
+        return self._url.name
+
+
+@dataclass
+class Get(Request):
+    method = "GET"
+    query: Union[str, Callable]
+
+    def __init__(self, query, **kwargs):
+        super().__init__(**kwargs)
+        self.query = query
+
+
+@dataclass
+class Post(Request):
+    method = "POST"
+    body: Union[str, Callable]
+
+    def __init__(self, body, **kwargs):
+        super().__init__(**kwargs)
+        self.body = body
+
+
+# Decorator
+def parametrize_response(param_values: list[Request], *, url_type: str = "NORMAL"):
```

Comment:
```suggestion
def parametrize_response(*param_values: Request, url_type: Url = Url.normal):
```

Dan heb je de `[ ... ]` niet meer nodig in alle tests, en geen vertaling van strings.  En anders gewoon wel de URL doorsturen?